doc/chapters/integration-guidelines/10-traffic-shaping.rst

```diff
@@ -0,0 +1,72 @@
+Linux offers a very rich set of tools for managing and manipulating the transmission of packets. And
+since the host system is Linux based, those tools can be used to manage and shape traffics on Software
+Containers. Linux community is very familiar with the tools available under Linux for packet mangling and
+firewalling (netfilter, and before that, ipchains) as well as hundreds of network services which can
+run on the operating system.
+
+Linux community are aware of the tremendous power of the traffic control subsystem which has grown
+and matured. To glace those variaty of solutions the linux documentation project would be very good
+resource to start. Please check `TLDP <http://www.tldp.org/HOWTO/html_single/Traffic-Control-HOWTO/#intro>`_
+for having understanding of the concept. `Linux Advanced Routing & Traffic Control <http://lartc.org/howto/>`_
+is another good source that presents deep information about the topic.
+
+However there are many ways of shaping the traffic, the purpose of this document is to give brief information
+with examples about one particular way of shaping network traffic of a container to prevent any possible
+network starvation scenario.
+
+This can be done via the network classifier cgroup. The Network classifier cgroup provides an interface
+to tag network packets with a class identifier. This way each software container produce network packages
+with different class id and tc can be used to assign different priorities to packets from different cgroups.
+
+The network classifier cgroup is supported to mark network packages by :ref:`CGroups gateway <cgroups-gateway>`.
+
+Example
+-------
+Here is an example way of marking network packages belongs to container.
+
+
+Create a service manifest to let container send/receive files on network and mark network classifier
+cgroup. Then add this manifest to default manifest folder :
+
+.. literalinclude:: examples/starvation_example.json
+
+
+Then, from the build directory, run the agent:
+
+.. literalinclude:: ../getting-started/examples/01_start_sc.sh
+
+
+Next, we will start a new container.
+
+
+Start a container:
+
+.. literalinclude:: ../getting-started/examples/03_create_container.sh
+
+
+Now we can set our prepared capabilities to mark network packages :
+
+.. literalinclude:: examples/setcapabilities.sh
+
+
+In this stage every network package within the container will be marked with class id 0x100001. This
+means every network package form this container will be marked with major class ID value 10 and minor
+class ID value 1. So it is wise to configure traffic shaping with ``tc`` (please check `tc components <http://www.tldp.org/HOWTO/html_single/Traffic-Control-HOWTO/#components>`_
+for more information about tc) according to indicated class IDs in the host ::
+
+
+# sudo tc qdisc add dev eth0 root handle 10: htb
+# sudo tc class add dev eth0 parent 10: classid 10:1 htb rate 1mbit
+# sudo tc filter add dev eth0 parent 10: protocol ip prio 10 handle 1: cgroup
```

Comment:
Actually, I'd prefer to have them in a formatting that is runnable as a script. I think this could all be scripted and run like the getting-started example.